runtime/lua/vim/fs.lua

```diff
@@ -227,7 +231,20 @@ end
 ---@return (string) Normalized path
 function M.normalize(path)
   vim.validate({ path = { path, 's' } })
-  return (path:gsub('^~/', vim.env.HOME .. '/'):gsub('%$([%w_]+)', vim.env):gsub('\\', '/'))
+  local p = (path:gsub('^~/', vim.env.HOME .. '/'):gsub('%$([%w_]+)', vim.env):gsub('\\', '/'))
+
+  -- Try to also convert the path to an absolute path with symlinks resolved, but if the path does
+  -- not yet exist simply return it as-is.
+  return M.realpath(p) or p
+end
+
+--- Convert a path to an absolute path, resolving symlinks.
+---
+---@param path (string) Path to convert to an absolute path
+---@return (string|nil) Absolute path, or nil if the file or directory does not
+---        exist.
```

Comment:
```suggestion
---@alias errmsg string|nil
---@alias errcode string|nil

----- Convert a path to an absolute path, resolving symlinks.
---
---@param path (string) Path to convert to an absolute path
---@return string|nil,errmsg,errcode Absolute path, or a `fail` tuple
---        if the file or directory does not exist.
```

Reply:
I think that `fs_realpath` requires a deeper look, since we're not handling all the cases where this function can fail.

_see https://github.com/isaacs/fs.realpath/blob/c1a831d877138d9ee66f4dfbf2a8a80e0a899268/index.js#L16-L22 for a potential solution_

from https://docs.libuv.org/en/v1.x/fs.html#c.uv_fs_realpath

> 
>  **Warning**:  This function has certain platform-specific caveats that were discovered when used in Node.
> 
>   * macOS and other BSDs: this function will fail with UV_ELOOP if more than 32 symlinks are
>     found while resolving the given path.  This limit is hardcoded and cannot be sidestepped.
>   * Windows: while this function works in the common case, there are a number of corner cases
>     where it doesn't:
>     - Paths in ramdisk volumes created by tools which sidestep the Volume Manager (such as ImDisk)
>       cannot be resolved.
>     - Inconsistent casing when using drive letters.
>     - Resolved path bypasses subst'd drives.
> 
>     While this function can still be used, it's not recommended if scenarios such as the
>     above need to be supported.
>     The background story and some more details on these issues can be checked [here](https://github.com/nodejs/node/issues/7726).